diff --git a/wisdem/ccblade/meson.build b/wisdem/ccblade/meson.build
index a0417e22..841159cd 100644
--- a/wisdem/ccblade/meson.build
+++ b/wisdem/ccblade/meson.build
@@ -6,11 +6,9 @@ bem_source = custom_target('bemmodule.c',
                             )
 
 py3.extension_module('_bem',
-                    'src/bem.f90',
-                bem_source,
-                fortranobject_c,
-                include_directories: inc_np,
-                dependencies : py3_dep,
-                subdir: 'wisdem/ccblade',
-		link_language: 'fortran',
-                install : true)
+                     'src/bem.f90',
+                     bem_source,
+                     dependencies : fortranobject_dep,
+                     subdir: 'wisdem/ccblade',
+		     link_language: 'fortran',
+                     install : true)
diff --git a/wisdem/meson.build b/wisdem/meson.build
index c4999da6..8b6a0194 100644
--- a/wisdem/meson.build
+++ b/wisdem/meson.build
@@ -11,6 +11,9 @@ if incdir_numpy == ''
 endif
 # this creates a raw string which is useful for Windows use of '\' for paths
 incdir_numpy = '''@0@'''.format(incdir_numpy)
+inc_np = include_directories(incdir_numpy)
+numpy_nodepr_api = ['-DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION']
+np_dep = declare_dependency(include_directories: inc_np, compile_args: numpy_nodepr_api)
 
 #incdir_f2py = run_command(py3,
 #    ['-c', 'import os; os.chdir(".."); import numpy.f2py; print(numpy.f2py.get_include())'],
@@ -18,11 +21,20 @@ incdir_numpy = '''@0@'''.format(incdir_numpy)
 #).stdout().strip()
 
 incdir_f2py = incdir_numpy / '..' / '..' / 'f2py' / 'src'
-inc_f2py = include_directories(incdir_f2py)
 fortranobject_c = incdir_f2py / 'fortranobject.c'
+inc_f2py = include_directories(incdir_f2py)
 
-inc_np = include_directories(incdir_numpy, incdir_f2py)
-
+# Share this object across multiple modules.
+fortranobject_lib = static_library('_fortranobject',
+  fortranobject_c,
+  c_args: numpy_nodepr_api,
+  dependencies: py3_dep,
+  include_directories: [inc_np, inc_f2py],
+)
+fortranobject_dep = declare_dependency(
+  link_with: fortranobject_lib,
+  include_directories: [inc_np, inc_f2py],
+)
 
 subdir('ccblade')
 subdir('pyframe3dd')
diff --git a/wisdem/pyframe3dd/meson.build b/wisdem/pyframe3dd/meson.build
index c25e19df..584cd7c5 100644
--- a/wisdem/pyframe3dd/meson.build
+++ b/wisdem/pyframe3dd/meson.build
@@ -1,26 +1,31 @@
 
-sources = ['src/py_main.c',
-     'src/py_structs.h',
-     'src/py_io.c',
-     'src/py_io.h',
-     'src/py_frame3dd.c',
-     'src/py_frame3dd.h',
-     'src/py_eig.c',
-     'src/py_eig.h',
-     'src/py_HPGmatrix.c',
-     'src/py_HPGmatrix.h',
-     'src/coordtrans.c',
-     'src/coordtrans.h',
-     'src/HPGutil.c',
-     'src/HPGutil.h',
-     'src/NRutil.c',
-     'src/NRutil.h']
+sources = [
+  'src/common.h',
+  'src/py_structs.h',
+  'src/HPGutil.h',
+  'src/HPGutil.c',
+  'src/coordtrans.h',
+  'src/coordtrans.c',
+  'src/NRutil.h',
+  'src/NRutil.c',
+  'src/py_io.h',
+  'src/py_io.c',
+  'src/py_eig.h',
+  'src/py_eig.c',
+  'src/py_HPGmatrix.h',
+  'src/py_HPGmatrix.c',
+  'src/py_frame3dd.h',
+  'src/py_frame3dd.c',
+  'src/py_main.c',
+]
 
 # We need to build a shared library NOT A PYTHON EXTENSION
 # The ctypes wrapper handles the extension part.
 # If the interface was done purely in C, then need the python extension.
-shared_library('_pyframe3dd', sources,
-     name_prefix: '',
-     include_directories: 'src',
-     dependencies : py3_dep,
-     install : true)
+shared_library(
+  '_pyframe3dd',
+  sources,
+  name_prefix: '',
+  include_directories: ['src','src/microstran'],
+  install : true,
+)
diff --git a/wisdem/rotorse/meson.build b/wisdem/rotorse/meson.build
index 06070f10..af01311d 100644
--- a/wisdem/rotorse/meson.build
+++ b/wisdem/rotorse/meson.build
@@ -6,11 +6,9 @@ precomp_source = custom_target('precompmodule.c',
                             )
 
 py3.extension_module('_precomp',
-                    'src/PreCompPy.f90',
-                precomp_source,
-                fortranobject_c,
-                include_directories: inc_np,
-                dependencies : py3_dep,
-                subdir: 'wisdem/rotorse',
-		link_language: 'fortran',
-                install : true)
+                     'src/PreCompPy.f90',
+                     precomp_source,
+                     dependencies : fortranobject_dep,
+                     subdir: 'wisdem/rotorse',
+		     link_language: 'fortran',
+                     install : true)
